nixos/modules/misc/nixpkgs.nix

```diff
@@ -42,12 +44,42 @@ let
     merge = lib.mergeOneOption;
   };
 
-  _pkgs = import ../../.. config.nixpkgs;
+  pkgsType = mkOptionType {
+    name = "nixpkgs";
+    description = "An evaluation of NixPkgs; the top level attribute set of packages";
+    check = builtins.isAttrs;
+  };
+
+  _pkgs =
+    if cfg.pkgs == null
+    then import ../../.. { inherit (cfg) config overlays system; }
+    else cfg.pkgs;
 
 in
 
 {
   options.nixpkgs = {
+
+    pkgs = mkOption {
+      default = null;
+      defaultText = "Situation-dependent; plain NixOS sets this to <code>null</code>.";
+      type = types.nullOr pkgsType;
+      example = literalExample ''import pkgs.path {}'';
```

Comment:
That's sounds like a nice example of an infinite loop, if someone write that within a module.
Maybe use `<nixpkgs>` instead of `pkgs.path`.